src/core/qgsprojectservervalidator.cpp

```diff
@@ -0,0 +1,129 @@
+/***************************************************************************
+                             qgsprojectservervalidator.cpp
+                             ---------------------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Etienne Trimaille
+    email                : etienne dot trimaille at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsapplication.h"
+#include "qgslayertreelayer.h"
+#include "qgsprojectservervalidator.h"
+#include "qgsvectorlayer.h"
+
+
+QString QgsProjectServerValidator::displayValidationError( QgsProjectServerValidator::ValidationError error )
+{
+  switch ( error )
+  {
+    case QgsProjectServerValidator::Encoding:
+      return QObject::tr( "Encoding is not set properly. It shouldn't be 'System'" );
```

Comment:
"It shouldn't be 'System'" doesn't sound professional. Maybe "An explicit, non "System" encoding is required"